### PR TITLE
More bg-color combination work

### LINKED ISSUE #2005: Inaccessible footer styles
## Problems
Customers are getting into trouble with their footer markup. Much of the filtered_html markup available has not been accounted for as well as common footer markup like `<address>`.

Until we have more guidance about what content should go into the footer area, we need to address the content that is already out there.

## Proposed Solution

Extend the same coverage as bg-black text paragraphs styles have locally in uids_base and address policy and the future of footer styles in uids.

### LINKED ISSUE #2074: Change footer block iowa link
## Problem

The same block iowa is going to different places. the header goes to uiowa.edu, the footer links to the site being viewed.

## Proposed solution
Change the footer link to go to uiowa.edu and possibly link the site name below the block iowa instead.

### LINKED ISSUE #2080: Collection accordion styling issues
Looks like the collection body for the accordion view mode and the headline and collection body for the detailed list view mode need a different background color to not clash with section background

Screen Shot 2020-08-24 at 4 39 08 PM
Screen Shot 2020-08-24 at 4 47 00 PM

### PR DESCRIPTION
**Pull request by joewhitsitt**
Short term solutions in uids_base for https://github.com/uiowa/uids/issues/242 and https://github.com/uiowa/uids/issues/262

Fixes #2074 
Fixes #2005
Fixes #2080

## To Test

Try out the blocks (cta, collections, card, text area, body, webform, accordion, event and the footer contact block) with different color backgrounds and the gray text area block option.

You will also notice that SiteImprove extension no longer complains about #2074.

### CODE REVIEW

**Comment by dnkemper** [comment]
super pumped to start testing this 👍

**Comment by dnkemper** [comment]
Okay. Will keep adding to this comment. Here's something I noticed:

1. If section background is black or brain pattern black then you can't see the text of a Call to action block without a background selected Screen Shot 2020-08-25 at 10 02 49 AM

1. Card good
1. Call to action good

1. Brain pattern black for event block - can't see text Screen Shot 2020-08-25 at 10 22 35 AM
1. Image gallery good
1. page title hero good
1. slider good
1. Separator/Spacer: Can't see when no background color or separator/spacer color selected. Can't see when separator and page are brain pattern background (naturally) but not sure if we should prevent this somehow or leave it? Screen Shot 2020-08-25 at 10 45 12 AMScreen Shot 2020-08-25 at 10 42 25 AM
1. statistic good
1. text block good

**Comment by joewhitsitt** [comment]
Looks like there is still work to be done. Thanks @dnkemper

**Comment by dnkemper** [comment]
@joewhitsitt It's SO much better already! You rock!

**Comment by joewhitsitt** [comment]
@bspeare can I get your opinion on how to handle Separator/Spacer for all of the different bg color options?

**Comment by pyrello** [comment]
We are attempting to deprecate the separator/spacer and remove it from all sites that are currently using it, so I would say it is not worth the effort.

**Comment by joewhitsitt** [comment]
Just rediscovered this: https://github.com/uiowa/uiowa/issues/1739 and how it relates to cards always having a white bg. Should cards be transparent? If so, more work needs to be done in the PR. I did [this](https://github.com/uiowa/uiowa/blob/886937384263315b801a6bb1ca910423b4ff24ab/docroot/themes/custom/uids_base/scss/components/card.scss) after a Slack conversation with @bspeare

**Comment by bspeare** [comment]
> Just rediscovered this: #1739 and how it relates to cards always having a white bg. Should cards be transparent? If so, more work needs to be done in the PR. I did [this](https://github.com/uiowa/uiowa/blob/886937384263315b801a6bb1ca910423b4ff24ab/docroot/themes/custom/uids_base/scss/components/card.scss) after a Slack conversation with @bspeare

@joewhitsitt I could see adding a style variant for a `bg--gray` to the card to solve @mark-bennett-uiowa issue.  But I don't want to have it adjust to every background color at this point.

**Comment by joewhitsitt** [comment]
- [x] Fix color issues related to body, webform

File: docroot/themes/custom/uids_base/scss/components/card.scss [hunk 1]
```diff
@@ -26,3 +26,15 @@
     flex: 1 1 100%;
   }
 }
+
+.layout-builder-block.card {
```
Comment (bspeare) [inline docroot/themes/custom/uids_base/scss/components/card.scss]: @joewhitsitt should we move this to `admin.scss`? This is just for the editor appearance, right?

Comment (joewhitsitt) [reply docroot/themes/custom/uids_base/scss/components/card.scss]: I thought I had to do this to ensure proper backgrounds and content colors regardless of the background. I can't remember if I did this based on the preview or the actual.

Comment (bspeare) [reply docroot/themes/custom/uids_base/scss/components/card.scss]: I think that class only appears on the edit screen.

Comment (bspeare) [reply docroot/themes/custom/uids_base/scss/components/card.scss]: I've noticed that issue while working on https://github.com/uiowa/uiowa/pull/2076.

Comment (joewhitsitt) [reply docroot/themes/custom/uids_base/scss/components/card.scss]: moved it. will test some more tomorrow when i work on webform support

**Comment by joewhitsitt** [comment]
I think this is in a reviewable state again. Tweaked card and added support for body and webform. Ignoring separator. I removed main navigation from the scope of this PR and opened up a separate issue to decide how to deal with it. https://github.com/uiowa/uids/issues/305

**Comment by mark-bennett-uiowa** [comment]
@joewhitsitt Going to post screenshots as I go for each block, background, alignment combination I can think of if thats okay. It should be the top image will show in LB, the second will be a saved LB and back in View mode. Ill post issues as I see them:
Here is CTA on a black section background (is there a padding/margin issue with the brain pattern black background block?):
Screen Shot 2020-08-28 at 4 01 27 PM
Screen Shot 2020-08-28 at 4 01 51 PM
Screen Shot 2020-08-28 at 4 05 16 PM
Screen Shot 2020-08-28 at 4 05 34 PM

**Comment by mark-bennett-uiowa** [comment]
Brain pattern is good and works as usual, changes text to black text on the whitish background. When I changed this to brain pattern black, everything looks good, except that a brain pattern black section background and the brain pattern black block background do not line up. Will suggest to avoid this combination to editors if they miss this.
Screen Shot 2020-08-28 at 4 14 35 PM
Screen Shot 2020-08-28 at 4 14 45 PM

**Comment by mark-bennett-uiowa** [comment]
Brain pattern reversed background AND Gold background works as expected, text changed to black on yellow background. Maybe change button background to white with black text here to match gold block background option?? This is an issue with both brain pattern reversed background and old background with no block background selected.
Screen Shot 2020-08-28 at 4 19 23 PM
Screen Shot 2020-08-28 at 4 23 13 PM

**Comment by mark-bennett-uiowa** [comment]
Gray section background with any colored CTA background blocks are working as expected.

**Comment by mark-bennett-uiowa** [comment]
Now cards, I have rewritten this to make it easier to follow:
**Upon Layout Builder save, all card backgrounds change to white on any section background color selected below.**

- Black section background AND Brain pattern black section background still shows unreadable black text with cards while in LB preview mode. White card background issue upon LB save.
- Brain pattern section background, Brain pattern reversed section background, Gold section background and Gray section background all show properly with black text and white button background in LB preview. White card background issue upon LB save.

**Comment by mark-bennett-uiowa** [comment]
Event block (pre-Rick Porter changes) are working as expected. Only small difference I see is the color of the block icon in the Brain pattern reversed section background and the Gold section background.
Screen Shot 2020-08-28 at 5 43 28 PM

**Comment by mark-bennett-uiowa** [comment]
Collection block is working as expected. Gray collection item accordion header background and white background of expandable region does not change across section background colors. Descriptive List text colors change properly with section background color changes.

**Comment by mark-bennett-uiowa** [comment]
Statistics and Text Areas are working as expected. Text color and bar colors (in statistics) change to contrasting colors as other blocks do. Good to go here.

**Comment by mark-bennett-uiowa** [comment]
TL;DR version of testing (@joewhitsitt):

- CTA title color is yellow on Black section background or Brain pattern black section background, but turns to white on Brain pattern black block background.
- Padding/margin change when adding a Brain pattern black CTA block background to any section background color (see screenshot above).
- Brain pattern block background on similar Brain pattern section background does not line up properly. Should recommend to editor to avoid doing this, rather than preventing this combination in code?
- CTA link button color is gold in transparent block on Brain pattern reversed or Gold section backgrounds. Adding a Gold block background turns this to the proper white button color.
- Upon Layout Builder save, all card backgrounds change to white on any section background color selected below.
- Black section background AND Brain pattern black section background still shows unreadable black text with cards while in LB preview mode. White card background issue upon LB save.
- Brain pattern section background, Brain pattern reversed section background, Gold section background and Gray section background all show properly with black text and white button background in LB preview. White card background issue upon LB save.
- Event block location icon is gray on Brain pattern reversed section background, but white on Gold section background.
- Collection block working as intended.
- Statistics block working as intended.
- Text area block working as intended.

**Comment by joewhitsitt** [comment]
Thanks for the in-depth review @mark-bennett-uiowa.

I will address any of the a11y concerns in this PR. I am not going to make design decisions beyond what is accessible. Padding wasn't in scope. Might need a follow-up issues to address padding and how each component should look with each option.

CTA seems to have a pattern that isn't consistent but all should meet contrast-ratio now which is the immediate goal.

Until a bg--gray option is allowed for card, Ben said that the card blocks should be white bg regardless of section background. I pushed a fix for preview.

**Review by dnkemper — APPROVED** [review]
